sdks/python/apache_beam/pipeline.py

```diff
@@ -157,6 +157,113 @@ def _root_transform(self):
     """Returns the root transform of the transform stack."""
     return self.transforms_stack[0]
 
+  def _remove_labels_recursively(self, applied_transform):
+    for part in applied_transform.parts:
+      if part.full_label in self.applied_labels:
+        self.applied_labels.remove(part.full_label)
+      if part.parts:
+        for part2 in part.parts:
+          self._remove_labels_recursively(part2)
+
+  def _replace(self, override):
+
+    assert isinstance(override, PTransformOverride)
+    matcher = override.get_matcher()
+
+    output_map = {}
+
+    class OutputVisitor(PipelineVisitor): # pylint: disable=used-before-assignment
+
+      def __init__(self, pipeline, applied_labels):
+        self.pipeline = pipeline
+        self._applied_labels = applied_labels
+
+      def _replace_if_needed(self, transform_node):
+        if matcher.match(transform_node):
+          replacement_transform = override.get_replacement_transform(
+              transform_node.transform)
+          inputs = transform_node.inputs
+          # We only support replacing single-input PTransforms.
+          assert len(inputs) == 1
+          transform_node.transform = replacement_transform
+          self.pipeline.transforms_stack.append(transform_node)
+
+          # Keeping the same label for the replaced node but recursively
+          # removing labels of child transforms since they will be replaced
+          # during the expand below.
+          self.pipeline._remove_labels_recursively(transform_node)
+
+          new_output = replacement_transform.expand(inputs[0])
+
+          # Recording updated outputs. This cannot be done in the same visor
+          # since if we dynamically update output type here, we'll run into
+          # errors when visiting child nodes.
+          output_map[transform_node.outputs[None]] = new_output
```

Comment:
Expand is not limited to returning a single PCollection (for example, it could return a tuple of PCollections). It could also be a DoFn with multiple outputs, in which case only the main one has "label" None. 

If we don't intend to handle the more general case yet, we should at least check.